app/registries/foreman/plugin.rb

```diff
@@ -567,6 +569,11 @@ def register_status_extension(&block)
       @status_extension = block
     end
 
+    def register_gettext_domain(domain)
+      @gettext_domain = domain
```

Comment:
Thoughts on using the ID by default:
```suggestion
    def register_gettext_domain(domain = nil)
      @gettext_domain = domain || @id.to_s
```

You could even use `register_gettext(domain: nil)`.